### PR TITLE
DHFPROD-4728: Adding roles and users to export entity instances

### PR DESCRIPTION
**Pull request by rahulvudutala**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by rahulvudutala** [comment]
Adding this PR so that tests added in the PR https://github.com/marklogic/marklogic-data-hub/pull/3862 run with the user having the hub-central-entity-exporter role

File: marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java [hunk 1]
```diff
@@ -371,6 +371,7 @@ public Path getCustomMappingFunctionsDir() {
         writeRoleFile(rolesDir, "data-hub-saved-query-reader.json");
         writeRoleFile(rolesDir, "data-hub-saved-query-writer.json");
         writeRoleFile(rolesDir, "hub-central-downloader.json");
+        writeRoleFile(rolesDir, "hub-central-entity-exporter.json");
```
Comment (akshaysonvane) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/impl/HubProjectImpl.java:374]: Can we test for this file?
`testInit()` in HubProjectTest already performs such a test.

**Comment by akshaysonvane** [comment]
Pinging @ryanjdew since he is coordinating the role/authority stuff and this might require an update to the data hub security documentation.

File: marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/security.sjs [hunk 2]
```diff
@@ -43,6 +43,10 @@ class Security {
       response.authorities.push('canManageSavedQuery');
     }
 
+    if (currentRoleNames.includes('hub-central-entity-exporter')) {
+      response.authorities.push('canExportEntityInstances');
```
Comment (rjrudin) [inline marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/security.sjs:47]: The middle tier endpoint for exporting should then have @Secured("canExportEntityInstances") on it. You can use setTestUserRoles to verify that a user with and without the role can access the endpoint.

Comment (rahulvudutala) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/security.sjs:47]: @akshaysonvane Can you add this in the PR https://github.com/marklogic/marklogic-data-hub/pull/3862 as you are working on adding the endpoints.

Comment (akshaysonvane) [reply marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/security.sjs:47]: Yes, I'll add that.

**Pull request merged**